### PR TITLE
Expose internal_network in os_floating_ip

### PR DESCRIPTION
**Pull request by yfried**
##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
os_floating_ip

##### ANSIBLE VERSION
ansible 2.3.0

##### SUMMARY
Shade project has finally exposed this argument so now this module
matches old quantum_floatingip module's capabilities.

Uses ``nat_destination`` term instead of "internal_network" to match shade
terminology. Adds ``private_network`` and ``internal_network`` as aliases

### CODE REVIEW

**Review by emonty — APPROVED** [review]
shipit

**Comment by emonty** [comment]
Oh - patch looks good - but version_added needs to be set to 2.3

**Comment by rcarrillocruz** [comment]
Since you are adding a new parameter that relies on a recent Shade version, you should make use of StrictVersion in case the user provides it to make sure the shade version supports it.
Have a look at this code:

https://github.com/ansible/ansible-modules-core/blob/devel/cloud/openstack/os_router.py#L308

**Comment by rcarrillocruz** [comment]
needs_revision

File: cloud/openstack/os_floating_ip.py [hunk 1]
```diff
@@ -144,12 +168,18 @@ def main():
     if not HAS_SHADE:
         module.fail_json(msg='shade is required for this module')
 
+    if (module.params['project'] and
```
Comment (emonty) [inline cloud/openstack/os_floating_ip.py]: I think you mean nat_destination

File: cloud/openstack/os_floating_ip.py [hunk 2]
```diff
@@ -144,12 +168,18 @@ def main():
...
+            StrictVersion(shade.__version__) <= StrictVersion('1.8.0')):
+        module.fail_json(msg="To utilize project, the installed version of"
```
Comment (emonty) [inline cloud/openstack/os_floating_ip.py]: Same here.

**Review by emonty — CHANGES_REQUESTED** [review]
Looks great - copy/paste typo issue.

**Review by emonty — APPROVED** [review]
shipit

**Comment by gregdek** [comment]
Thanks @yfried-redhat for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by yfried** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @yfried-redhat. To the current maintainers, @emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by rcarrillocruz** [comment]
shipit

**Comment by abadger** [comment]
Thanks @yfried-redhat for the feature, @rcarrillocruz and @emonty for reviewing!  Merged to devel for the 2.3.0 release.

**Comment by pythondz** [comment]
Hello,

ansible 2.3.2.0
shade 1.22.0

I'm having trouble with modifications introduced by this PR :

Module is not idempotent and fail when a floating_ip is already assigned to a resource.

The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_LbNztb/ansible_module_os_floating_ip.py", line 276, in <module>
    main()
  File "/tmp/ansible_LbNztb/ansible_module_os_floating_ip.py", line 215, in main
    (nat_destination and f_ip.internal_network == fixed_address),
  File "/usr/local/lib/python2.7/dist-packages/munch/__init__.py", line 103, in __getattr__
    raise AttributeError(k)
AttributeError: internal_network

According to OpenStack network API, not such attribute is returned: https://developer.openstack.org/api-ref/network/v2/index.html#show-floating-ip-details

FWIW, I dump the f_ip object and internal_network does not exist : 

{
	'status': u'ACTIVE',
	'router_id': u'bd2e5daa-17b3-4edf-b085-e9325540dd46',
	'properties': Munch({
		
	}),
	'description': '',
	'tenant_id': u'b726dc6afc784c4d9c0a1fb4fe87e856',
	'created_at': None,
	'attached': True,
	'updated_at': None,
	'id': u'e6f69050-1c27-469e-bbaf-556096df048f',
	'floating_network_id': u'17bb9761-36be-457e-b57c-178430aed5ad',
	'fixed_ip_address': u'<hidden>',
	'floating_ip_address': u'<hidden>',
	'location': Munch({
		'project': Munch({
			'domain_id': None,
			'id': u'b726dc6afc784c4d9c0a1fb4fe87e856',
			'name': None,
			'domain_name': None
		}),
		'zone': None,
		'region_name': '',
		'cloud': 'envvars'
	}),
	'revision_number': None,
	'router': u'bd2e5daa-17b3-4edf-b085-e9325540dd46',
	'project_id': u'b726dc6afc784c4d9c0a1fb4fe87e856',
	'port_id': u'9f5a8581-4ec2-4487-a31b-e7f6b4357cd2',
	'port': u'9f5a8581-4ec2-4487-a31b-e7f6b4357cd2',
	'network': u'17bb9761-36be-457e-b57c-178430aed5ad'
}

Thanks

**Pull request merged**